### ISSUE TITLE
Add Matrix4x3

### ISSUE THREAD

**Issue by pjcozzi** [CesiumGS/cesium#2360]
This would be similar to GLSL's `mat4x3`: 4 columns and 3 rows.

Given that the vast majority of our `Matrix4` instances are transformation matrices, the bottom row is always `[0.0, 0.0, 0.0, 1.0]`.  `Matrix4x3` (or perhaps less generically, `TransformMatrix4`) would implicitly have this bottom row so it:
- Uses less memory and leads to less cache pollution.
- Is faster to copy (`clone`) and compare (`equals`).
- Has all math functions optimized by default, e.g., like [`Matrix4.multiplyTransformation`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Matrix4.js#L1461)
- Would interop with `Matrix4`.

This is a micro-optimization that would perhaps shave 0.5-3% off typical profiles since `Matrix4` functions are used everywhere.  For example, in the scene below, `clone` and `equals` take up almost 4% of the profile!

![image](https://cloud.githubusercontent.com/assets/782098/5578788/73853ae6-8ffc-11e4-8e1c-65c2a8915b3a.png)

**Comment by vu23h** [comment]
@pjcozzi Hi Patrick, my name is Huong and I'm a current MCIT student at Penn. I was introduced by Dylan to work on this issue and was wondering if you could help me get started since I am very new to open source projects. Thank you very much!

**Comment by pjcozzi** [comment]
@vu23h welcome!!!  Please read through the materials linked from the [Getting Started Contributing](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#getting-started-contributing) section.  Then, I suggest finding a few really simple issues labeled [beginner](https://github.com/AnalyticalGraphicsInc/cesium/labels/beginner) before starting on this one.  Once you complete that, we'll identity someone on the team to mentor you.

**Comment by ggetz** [comment]
This is unlikely to become a priority anytime soon. We'd still happily accept a PR addressing this issue. But tracking this isn't particularly useful because of its low priority, so I will close it. Thanks!
